server/sonar-server/src/main/java/org/sonar/server/webhook/WebhookQGChangeEventListener.java

```diff
@@ -49,20 +53,37 @@ public void onIssueChanges(QGChangeEvent qualityGateEvent, Set<ChangedIssue> cha
     if (!webhooks.isEnabled(qualityGateEvent.getProjectConfiguration())) {
       return;
     }
+    Optional<EvaluatedQualityGate> evaluatedQualityGate = qualityGateEvent.getQualityGateSupplier().get();
+    if (isQGStatusUnchanged(qualityGateEvent, evaluatedQualityGate)) {
+      return;
+    }
 
     try (DbSession dbSession = dbClient.openSession(false)) {
-      callWebhook(dbSession, qualityGateEvent);
+      callWebhook(dbSession, qualityGateEvent, evaluatedQualityGate.orElse(null));
+    }
+  }
+
+  private static boolean isQGStatusUnchanged(QGChangeEvent qualityGateEvent, Optional<EvaluatedQualityGate> evaluatedQualityGate) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Specify a "EvaluatedQualityGate" parameter instead. ([squid:S3553](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10247%2Fwebhooks_called_only_if_qg_status_changes&issues=AWEEZ0cXEFfjxkQcQAXs&open=AWEEZ0cXEFfjxkQcQAXs)

[ISSUE_KEY]: AWEEZ0cXEFfjxkQcQAXs

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED

Reply:
This issue has been reopened

[ISSUE_STATUS]: REOPENED

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED